filesystem/File.php

```diff
@@ -390,7 +390,6 @@ public function getCMSFields() {
 
 		//get a tree listing with only folder, no files
 		$folderTree = new TreeDropdownField("ParentID", _t('AssetTableField.FOLDER','Folder'), 'Folder');
-		$folderTree->setChildrenMethod('ChildFolders');
```

Comment:
Yeah it's a bit of a tricky one... the whole approach to generating and marking a tree could probably use a good look over. Thanks for the merge!